node/channel.js

```diff
@@ -453,20 +454,37 @@ TChannel.prototype.makeSubChannel = function makeSubChannel(options) {
     return chan;
 };
 
-TChannel.prototype.listen = function listen(port, host, callback) {
+TChannel.prototype.listen = function listen(port, host, options, callback) {
     // Note:
     // - 0 is a valid port number, indicating that the system must assign an
     //   available ephemeral port
     // - 127.0.0.1 is a valid host, primarily for testing
+    // - options supports:
+    //   {fd: X} - listen will attach to the existing socket, and host/port
+    //   remain required, but become only descriptive. See also:
+    //   https://nodejs.org/api/net.html#net_server_listen_handle_callback
     var self = this;
+    if (options && typeof options === 'function') {
+        callback = options;
+        options = {};
+    }
+    options = options || {};
     assert(!self.listened, 'TChannel can only listen once');
     assert(typeof host === 'string', 'TChannel requires host argument');
     assert(typeof port === 'number', 'TChannel must listen with numeric port');
+    assert(options.fd === undefined || typeof options.fd === 'number', 'TChannel listen options.fd must be numeric');
```

Comment:
switching to a hybrid solution, since undefined is legit, as is a positive number:

  `options.fd === undefined || (typeof options.fd === 'number' && options.fd >= 0)`

seems the most sane. And agreed on `NaN` annilhilation, which is why I like the >= test as well.